src/model/CoilUserDefined.cpp

```diff
@@ -0,0 +1,589 @@
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "CoilUserDefined.hpp"
+#include "CoilUserDefined_Impl.hpp"
+#include "Model.hpp"
+#include "EnergyManagementSystemProgramCallingManager.hpp"
+#include "EnergyManagementSystemProgramCallingManager_Impl.hpp"
+#include "EnergyManagementSystemProgram.hpp"
+#include "EnergyManagementSystemProgram_Impl.hpp"
+#include "EnergyManagementSystemActuator.hpp"
+#include "EnergyManagementSystemActuator_Impl.hpp"
+#include "ThermalZone.hpp"
+#include "ThermalZone_Impl.hpp"
+
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_Coil_UserDefined_FieldEnums.hxx>
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+  namespace detail {
+
+    CoilUserDefined_Impl::CoilUserDefined_Impl(const IdfObject& idfObject, Model_Impl* model, bool keepHandle)
+      : WaterToAirComponent_Impl(idfObject, model, keepHandle) {
+      OS_ASSERT(idfObject.iddObject().type() == CoilUserDefined::iddObjectType());
+    }
+
+    CoilUserDefined_Impl::CoilUserDefined_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model, bool keepHandle)
+      : WaterToAirComponent_Impl(other, model, keepHandle) {
+      OS_ASSERT(other.iddObject().type() == CoilUserDefined::iddObjectType());
+    }
+
+    CoilUserDefined_Impl::CoilUserDefined_Impl(const CoilUserDefined_Impl& other, Model_Impl* model, bool keepHandle)
+      : WaterToAirComponent_Impl(other, model, keepHandle) {}
+
+    const std::vector<std::string>& CoilUserDefined_Impl::outputVariableNames() const {
+      static std::vector<std::string> result;
+      if (result.empty()) {
+      }
+      return result;
+    }
+
+    std::vector<openstudio::IdfObject> CoilUserDefined_Impl::remove() {
+      if (isRemovable()) {
+        return WaterToAirComponent_Impl::remove();
+      }
+      return {};
+    }
+
+    IddObjectType CoilUserDefined_Impl::iddObjectType() const {
+      return CoilUserDefined::iddObjectType();
+    }
+
+    ModelObject CoilUserDefined_Impl::clone(Model model) const {
+      auto newCoilUserDefined = ModelObject_Impl::clone(model).cast<CoilUserDefined>();
+      //airOutletTemperatureActuator
```

Comment:
This clone method looks suspicious to me:
* I was kinda expecting you to clear out some of XXX Node Name field, as I thought this object had more connections than the WaterToAirComponent base class (air/water)
* You're listing all these objects as children, so I guess they should be cloned automagically if you didn't call ModelObject_Impl::clone which bypasses ParentObject_Impl::clone

I'm going to be paying close attention to the model tests as a result.